classes/OccurrenceManager.php

```diff
@@ -638,7 +638,10 @@ public function getQueryTermStr(){
 			if($v) $retStr .= '&'. htmlspecialchars($this->cleanOutStr($k), ENT_QUOTES) . '=' . htmlspecialchars($this->cleanOutStr($v), ENT_QUOTES);
 		}
 		if(isset($this->taxaArr['search'])){
-			$retStr .= '&taxa=' . htmlspecialchars($this->cleanOutStr($this->taxaArr['search']), ENT_QUOTES);
+			$patternOfOnlyLettersDigitsAndSpaces = '/^[a-zA-Z0-9\s]*$/';
```

Comment:
Does this regex allow just whitespaces? And does it cause any issues?

Reply:
It does allow just whitespace. The behavior seems to be in that case that all taxa are searched across. These seems like a sane thing to do? At least just blank spaces aren't exploitable?

Reply:
may need to include hyphen: https://biology.stackexchange.com/questions/108056/animal-specific-names-with-non-alphabetic-characters

Reply:
Fixed in 2dbee8bca. Note: this may not be the final solve for this vulnerability. Added a TODO comment for additional edge cases as they come up.